lib/auth/grpcserver.go

```diff
@@ -246,6 +246,42 @@ func (g *GRPCServer) SetAccessRequestState(ctx context.Context, req *proto.Reque
 	return &empty.Empty{}, nil
 }
 
+func (g *GRPCServer) GetPluginData(ctx context.Context, filter *services.PluginDataFilter) (*proto.PluginDataSeq, error) {
+	// TODO(fspmarshall): Add rate-limiting.
+	auth, err := g.authenticate(ctx)
+	if err != nil {
+		return nil, trail.ToGRPC(err)
+	}
+	data, err := auth.AuthWithRoles.GetPluginData(ctx, *filter)
+	if err != nil {
+		return nil, trail.ToGRPC(err)
+	}
+	var seq []*services.PluginDataV3
```

Comment:
why pointers here? You can use `nullable` modifier to prevent grpc from generating arrays of pointers, see examples:

https://github.com/gravitational/teleport/blob/master/lib/services/types.proto#L35

Reply:
From what I can tell, this is consistent with how we handle `Resource` types elsewhere in the GRPC API.  At a guess, this is because we cast all our resource types to interfaces anyhow.